### ISSUE TITLE
Wrong error message for use before def in an enum

### ISSUE THREAD

**Issue by mhegazy** [Microsoft/TypeScript#4435]
```ts
 enum DeclKind {
    Variable = 1,
    SomeValue = Field,
    Field = Variable,
}
```
> Error TS2651: A member initializer in a 'const' enum declaration cannot reference members declared after it, including members defined in other 'const' enums.
